plugin.py

```diff
@@ -108,47 +108,51 @@ def install_or_update(cls) -> None:
             raise
 
     def on_pre_server_command(self, command: Mapping[str, Any], done_callback: Callable[[], None]) -> bool:
-
-        cargo_commands = []
-        for c in command["arguments"]:
-            if c["kind"] == "cargo":
-                cargo_commands.append(c)
-
-        if len(cargo_commands) == 0:
-            return False
-
-        window = sublime.active_window()
-        if window is None:
-            return False
-        view = window.active_view()
-        if not view:
-            return False
-        if not Terminus:
-            sublime.error_message(
-                'Cannot run executable. You need to install the "Terminus" package and then restart Sublime Text')
-            done_callback()
-            return True
-        if not shutil.which("cargo"):
-            sublime.error_message('Cannot find "cargo" on path.')
-            return False
-        main_cargo_path = '"{}"'.format(shutil.which("cargo"))
-        for output in cargo_commands:
-            if output["args"]["overrideCargo"]:
-                cargo_path = output["args"]["overrideCargo"]
-            else:
-                cargo_path = main_cargo_path
-            command_to_run = [cargo_path] + output["args"]["cargoArgs"] + \
-                output["args"]["cargoExtraArgs"]
-            cmd = " ".join(command_to_run)
-            args = {
-                "title": output["label"],
-                "shell_cmd": cmd,
-                "cwd": output["args"]["workspaceRoot"],
-                "auto_close": get_setting(view, "terminus_auto_close", False)
-            }
-            if get_setting(view, "terminus_use_panel", False):
-                args["panel_name"] = output["label"]
-            window.run_command("terminus_open", args)
+        if command["command"].startswith("rust-analyzer"):
```

Comment:
Can you tell us what are the specific command names that are triggered for those "cargo" commands? A `rust-analyzer` prefix still sounds quite generic. Maybe too generic.

BTW. Early return would avoid an extra level of indentation for the whole method.

Reply:
Okay so the command can be one of two
a) rust-analyzer.runSingle
b) rust-analyzer.runDebug

With this code both would be handled!

Reply:
Yeah but don't use `startswith`. Handle the specific commands that you want to handle. I've shown example where it goes wrong with `startswith`.